Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -301,84 +265,61 @@ void FullscreenManager::cancelFullscreen()
 
     INFO_LOG(LOGIDENTIFIER);
 
-    // To achieve that aim, remove all the elements from the top document's stack except for the first before
-    // calling webkitExitFullscreen():
-    RefPtr fullscrenElement = topDocument.fullscreenManager().fullscreenElement();
-    topDocument.fullscreenManager().m_fullscreenElementStack = { WTFMove(fullscrenElement) };
+    m_pendingExitFullscreen = true;
 
-    topDocument.fullscreenManager().exitFullscreen();
+    m_document.eventLoop().queueTask(TaskSource::MediaElement, [this, &topDocument, identifier = LOGIDENTIFIER] {
+        if (!topDocument.page()) {
+            INFO_LOG(LOGIDENTIFIER, "Top document has no page.");
+            return;
+        }
+
+        // This triggers finishExitFullscreen with ExitMode::Resize, which fully exits the document.
+        if (auto* fullscreenElement = topDocument.fullscreenManager().fullscreenElement())
+            page()->chrome().client().exitFullScreenForElement(fullscreenElement);
```

Comment:
Seems like `exitFullScreenForElement` should take a reference, not a pointer. Something to refine in the future, I guess.